### PR TITLE
bump next.js to 12.x

### PR DESCRIPTION
**Pull request by Yeti-or**
:baby_chick: Download canary assets:

[colors--canary.371.4290853494.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.371.4290853494.xml)  
[color_metro_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.371.4290853494.swift)  
[color_metro_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.371.4290853494.kt)  
[color_metro_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.371.4290853494.ts)  
[color_sberHealth_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.371.4290853494.swift)  
[color_sberHealth_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.371.4290853494.kt)  
[color_sberHealth_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.371.4290853494.ts)  
[color_sbermarket_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.371.4290853494.swift)  
[color_sbermarket_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.371.4290853494.kt)  
[color_sbermarket_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.371.4290853494.ts)  
[color_sberprime_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.371.4290853494.swift)  
[color_sberprime_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.371.4290853494.kt)  
[color_sberprime_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.371.4290853494.ts)  
[color_selgros_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.371.4290853494.swift)  
[color_selgros_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.371.4290853494.kt)  
[color_selgros_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.371.4290853494.ts)  
[color_smbusiness_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.371.4290853494.swift)  
[color_smbusiness_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.371.4290853494.kt)  
[color_smbusiness_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.371.4290853494.ts)  
[PlasmaTokensColor--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.371.4290853494.swift)  
[shadow_sbermarket_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.371.4290853494.ts)  
[typo_mage_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.371.4290853494.swift)  
[typo_mage_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.371.4290853494.kt)  
[typo_mage_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.371.4290853494.ts)  
[typo_plasma_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.371.4290853494.swift)  
[typo_plasma_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.371.4290853494.kt)  
[typo_plasma_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.371.4290853494.ts)  
[typo_ruler_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.371.4290853494.swift)  
[typo_ruler_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.371.4290853494.kt)  
[typo_ruler_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.371.4290853494.ts)  
[typo_sage_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.371.4290853494.swift)  
[typo_sage_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.371.4290853494.kt)  
[typo_sage_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.371.4290853494.ts)  
[typo_sbermarket_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.371.4290853494.swift)  
[typo_sbermarket_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.371.4290853494.kt)  
[typo_sbermarket_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.371.4290853494.ts)  
[typo_soulmate_ios-swift--canary.371.4290853494.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.371.4290853494.swift)  
[typo_soulmate_kotlin--canary.371.4290853494.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.371.4290853494.kt)  
[typo_soulmate_react-native--canary.371.4290853494.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.371.4290853494.ts)

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-371/

**Comment by Yakutoc** [comment]
@Yeti-or Привет pipeline `PR Documentation and Storybook / Build` упал с той же ошибкой что у меня 

```ts
./components/GlobalStyle.tsx:40:37
Type error: Argument of type 'InterpolationFunction<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }>' is not assignable to parameter of type 'TemplateStringsArray | CSSObject | InterpolationFunction<ThemedStyledProps<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }, DefaultTheme>>'.
  Type 'InterpolationFunction<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }>' is not assignable to type 'InterpolationFunction<ThemedStyledProps<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }, DefaultTheme>>'.
    Type 'Interpolation<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }>' is not assignable to type 'Interpolation<ThemedStyledProps<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }, DefaultTheme>>'.
      Type 'InterpolationFunction<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }>' is not assignable to type 'Interpolation<ThemedStyledProps<{ deviceScale?: number | undefined; breakWord?: boolean | undefined; }, DefaultTheme>>'.

  38 |     }
  39 | `;
> 40 | const TypoStyle = createGlobalStyle(standard);
     |                                     ^
  41 | 
  42 | export const GlobalStyle = () => {
  43 |     const { state } = useContext(Context);
```

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-371/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-371/

**Comment by Yeti-or** [comment]
@Yakutoc похоже я победил

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-371/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-371/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-371/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-371/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-371/temple-storybook/

**Comment by Yakutoc** [comment]
@Yeti-or Тогда твое решение вольем. 

Странно, у меня почти такое же решение - не прошло. 

https://github.com/salute-developers/plasma/pull/369/commits/5c39fc046efbbce853891a44b942ec86a7919c6f 

```
        "next": "^11.1.2",
        "react": "17.0.2",
        "react-dom": "17.0.2",
        -"styled-components": "^5.3.0"
        +"styled-components": "^5.3.1"
    },
    "devDependencies": {
        "@types/react": "17.0.11",
        "@types/react-dom": "^17.0.8",
       -"@types/styled-components": "^5.1.10",
       +"@types/styled-components": "^5.1.0",
        "eslint": "7.29.0",
        "eslint-config-next": "11.0.1",
        "typescript": "4.3.4"
```

**Comment by Yeti-or** [comment]
@Yakutoc ох надеюсь не огребем когда будем бампать версии styled-components =/

**Comment by Yakutoc** [comment]
> @Yakutoc ох надеюсь не огребем когда будем бампать версии styled-components =/

Судя по всему - по шапке получим ) 

Уже какие-то ошибки вылазят (

**Review by Yakutoc — APPROVED** [review]

File: website/plasma-website/pages/_document.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default class Document extends BaseDocument {
                         href="https://cdn-app.sberdevices.ru/shared-static/0.0.0/styles/SBSansDisplay.0.1.0.css"
                     />
                     <link rel="preconnect" href="https://fonts.googleapis.com" />
-                    <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="true" />
+                    <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="" />
```
Comment (Yakutoc) [inline website/plasma-website/pages/_document.tsx:48]: @Yeti-or А почему так сделали?

Comment (Yeti-or) [reply website/plasma-website/pages/_document.tsx:48]: @neretin-trike может ты лучше знаешь что здесь должно быть?

Comment (neretin-trike) [reply website/plasma-website/pages/_document.tsx:48]: честно говоря тоже не совсем понял. Я бы оставил как было (с `true`), если оно не ломается

Comment (neretin-trike) [reply website/plasma-website/pages/_document.tsx:48]: а вообще судя по [доке](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attribute), тут `true` и не может быть. Либо `anonymous`, либо `use-credentials`, ибо пустая

File: examples/demo-canvas-app/package.json [hunk 2]
```diff
@@ -23,14 +23,14 @@
...
-        "next": "11.0.1",
+        "next": "^12.3.4",
         "react": "17.0.2",
         "react-dom": "17.0.2",
         "styled-components": "^5.3.0"
     },
     "devDependencies": {
-        "@types/react": "17.0.12",
-        "@types/styled-components": "^5.1.10",
+        "@types/react": "^18.0.28",
```
Comment (neretin-trike) [inline examples/demo-canvas-app/package.json:32]: а я правильно понимаю, что мы можем вливать эту только после https://github.com/salute-developers/plasma/pull/194?

Comment (Yeti-or) [reply examples/demo-canvas-app/package.json:32]: Не, можно прям сейчас

Comment (neretin-trike) [reply examples/demo-canvas-app/package.json:32]: т.к. на этой версии всё уже нормально работает?

Comment (Yeti-or) [reply examples/demo-canvas-app/package.json:32]: Да мы же работаем и на 17 и на 18, некст просит новую версию тайпингов, для обновления;

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**